transforms/csdl/preprocess_csdl.xsl

```diff
@@ -634,4 +634,38 @@
       <xsl:apply-templates select="node()"/>
     </xsl:copy>
   </xsl:template>
+
+  <!-- Add inner error description -->
+  <xsl:template match="edm:Schema[@Namespace='microsoft.graph']">
+    <xsl:copy>
+      <xsl:apply-templates select="@* | node()"/>
+      <xsl:element name="ComplexType">
+        <xsl:attribute name="Name">InnerError</xsl:attribute>
+        <xsl:element name="Property">
+          <xsl:attribute name="Name">request-id</xsl:attribute>
```

Comment:
thanks, I was afraid of that, I'm not sure who designed the inner error payload back then ignoring every convention...
Ideally we don't want this type in the TypeWriter based SDK, since it's already assumed in the generation process. I was thinking about adding a super custom attribute on the type (like TypeWriter:Ignore="true") and making a dirty patch over there to skip it. Thoughts?

Reply:
Can we add a parameter similar to:
https://github.com/microsoftgraph/msgraph-metadata/blob/fd8d068cd284a5693be9d1052d5f91d0b54211d1/transforms/csdl/preprocess_csdl.xsl#L7

and then maybe produce two different clean metadata versions in the generation process, one for OpenAPI based SDKs and one for Typewriter SDKs.

I am OK with a temporary hack idea as well since Typewriter is going away, but the param approach could arguably take the same amount of time to implement.

Reply:
good suggestion!
I just pushed a commit to add the parameter, let me know what you think.
Now the question I have is how to we "fork the description" in the generation pipeline? different build artifacts and run the scripts twice with different argument sets?

Reply:
That's what I had in mind.

The only addition I have is that maybe we can call the parameter something like `transform-for-openapi` instead of making it specific to generation of error descriptions. That will give us the reusability of the same parameter for all other scenarios where the fix is OpenAPI specific. We can postpone this until we have additional scenarios come in too.

Reply:
I'd suggest we postpone it until we have additional use cases (like contains target), at least now the name of the parameter is explicit to what it does. resolving.